main.tf

```diff
@@ -44,6 +45,11 @@ resource "aws_apigatewayv2_domain_name" "this" {
     security_policy = "TLS_1_2"
   }
 
+  mutual_tls_authentication {
```

Comment:
Probably `mutual_tls_authentication` should be written with `dynamic "mutual_tls_authentication"` to make this module work in both cases - with mTLS and without it.